File: SEP3_CSharp/Entities/Order.cs
using System.Collections;

namespace Entities;

public class Order
{
    public int OrderId { get; set; }
    public double Price { get; set; }
    public DateTime PlacedOn { get; set; }
    public PaymentMethod PaymentMethod { get; set; } = null!;
    public string PlacedBy { get; set; }  
    public DeliveryOption DeliveryOption { get; set; }

    public List<Item> Items { get; set; }

    public bool IsNew()
    {
        DateTime temp = PlacedOn.AddDays(30);
        if (DateTime.Compare(DateTime.Now, temp) > 0)
        {
            return false;
        }
        else
        {
            return true;
        }
    }

}

File: SEP3_Java/SEP3_Java/src/main/java/Entities/Order.java
package Entities;

import Entities.Utlities.PaymentMethod;

import java.time.LocalDate;
import java.util.ArrayList;

public class Order
{
  private int orderId;
  private LocalDate placedOn;
  private PaymentMethod paymentMethod;
  private double totalAmount;
  private int deliveryOption; //This one may change
  private ArrayList<Item> items;
  private String placedBy;

  public Order(ArrayList<Item> items, double amount)
  {
    this.items = items;
    totalAmount = amount;
  }

  public ArrayList<Item> getItems()
  {
    return items;
  }

  public double getTotalAmount()
  {
    return totalAmount;
  }

  public LocalDate getDate()
  {
    return placedOn;
  }

  public int getDeliveryOption()
  {
    return deliveryOption;
  }

  public PaymentMethod getPaymentMethod()
  {
    return paymentMethod;
  }

  public void setDeliveryOption(int deliveryOption)
  {
    this.deliveryOption = deliveryOption;
  }
  public void setPaymentMethod(PaymentMethod paymentMethod)
  {
    this.paymentMethod = paymentMethod;
  }

  public void setDate(LocalDate date)
  {
    this.placedOn = date;
  }

}
